### ISSUE TITLE
'dsym_path' value must be a String! Found Array instead.

### ISSUE THREAD

**Issue by EmDee** [bugsnag/bugsnag-dsym-upload#17]
### Expected behavior
Being able to pass an array with multiple locations to `dSYM` files.

### Observed behavior
Error stating `'dsym_path' value must be a String! Found Array instead.`.

### Steps to reproduce
Run a lane with `upload_symbols_to_bugsnag(dsym_path: ['./Carthage/Build/iOS/', './artifacts/'])`.

### Version
`1.3.2`

### Additional information
According to the docs and the release notes, we should be able to pass an array to the `dsym_path` option.

**Comment by Cawllec** [comment]
Hi @EmDee, thanks for the report, I'll look into this.

**Comment by Cawllec** [comment]
I've had a look into this and it appears that we are missing the config `type: Array` option in the `:dsym_path` option, which is preventing arrays from being used as indicated.  I'll get started on a fix right away!

**Comment by EmDee** [comment]
Ok, then I misinterpreted the `Path to the DSYM directory[...]` description. Directory seems a bit ambiguous here. Maybe `dSYM package` or even `dSYM file` might be more accurate.

One more question: If I pass a `zip`, can it include multiple `dSYM` files or does it have to be a single `dSYM` per `zip`?

**Comment by Cawllec** [comment]
Sorry, I misread what was happening.
The two validation steps are:
1. Does the file exist?
2. Is the file either a directory, or ending in `.zip` or `.dSYM`
So if you pass a directory that exists, that _should_ be working.

**Comment by Cawllec** [comment]
The structure of the `zip` should match that of the zip files downloadable from iTunes connect.  This contains multiple dSYMs, so you should be ok with multiple in one `zip`.

**Comment by EmDee** [comment]
> So if you pass a directory that exists, that should be working.

Then there seems to be a bug there as well.
When I try `upload_symbols_to_bugsnag(dsym_path: "test")` where `test` is a folder, then I get `Failed uploading test` error (also tried with `test/`).

**Comment by Cawllec** [comment]
I'll have another look into that quickly

**Comment by Cawllec** [comment]
That sounds like an issue in our upload script, could you run `upload_symbols_to_bugsnag` with `verbose: true`?

**Comment by EmDee** [comment]
It looks like when passing a directory, it only scans for `*.dSYM` and ignores `zip` and `DWARF` files. Here's the output of where only a `zip` file is inside of `test`:

```
Uploading files to https://upload.bugsnag.com
Preparing to upload test/*.dSYM
Skipping file missing DWARF data: test/*.dSYM
1 files failed to upload
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios dsym |
+------------------+----------+
[15:07:27]: Failed uploading test
```

And here's the output with a correct `dSYM`, the `DWARF` file, and the `zip` inside

```
Uploading files to https://upload.bugsnag.com
Preparing to upload test/MapboxStatic.framework.dSYM
Uploading UUID: 9784D45B-81CF-3608-B546-A7258F550D0D (i386) test/MapboxStatic.framework.dSYM/Contents/Resources/DWARF/MapboxStatic UUID: 1304C1E1-CB25-3080-ACC8-EE3ED3766AAD (x86_64) test/MapboxStatic.framework.dSYM/Contents/Resources/DWARF/MapboxStatic UUID: 0EDBD725-2F38-3947-B5C8-ECA724A21843 (armv7) test/MapboxStatic.framework.dSYM/Contents/Resources/DWARF/MapboxStatic UUID: AC4B4798-818B-3D87-B759-16346F0D95A4 (arm64) test/MapboxStatic.framework.dSYM/Contents/Resources/DWARF/MapboxStatic

1 files uploaded successfully
```

**Comment by Cawllec** [comment]
Ok, I'm not especially knowledgeable about the contents of dSYMs or how we manage the upload, but I'll verify if this is the behaviour we want and expect to be happening with someone who knows a bit more.

**Comment by Cawllec** [comment]
So our validation process before uploading is:
- First ensure there is a UUID labeled dSYM folder (i.e. `01234567-89AB-CDEF-0123-456789ABCDEF.dSYM`)
- Ensure the `DWARF` folder exists in the correct place (i.e. `*.dSYM/Contents/Resource/DWARF`)
And then the underlying dSYM within the `DWARF` folder is uploaded.
The process should be similar for a `.zip`, but the `.zip` file needs to be explicitly identified.

**Comment by Cawllec** [comment]
Yes, on checking we're actually testing the UUID in the dSYM under the `DWARF` folder, rather than the top-level generated folder.  However, the `DWARF` folder validation (i.e. in the `Contents/Resource/DWARF` position) is still being made.

**Comment by EmDee** [comment]
What does this mean practically?

If I pass a directory as `dsym_path` it must contain the `dSYM` folder and cannot contain a zip or the actual `dSYM` file?

**Comment by Cawllec** [comment]
If you want to use a `.zip` it should be provided directly, and all `dSYM`s to be uploaded should be inside it.  If you provide a folder it will only upload `*.dSYM` folders, ignoring all other files/folders.

**Comment by EmDee** [comment]
Ok, cool. Then only the issue with passing an array needs to be fixed here.

The documentation of the plugin would benefit a lot with the clarification from this issue here.

**Comment by kattrali** [comment]
Array issues fixed in v1.3.3
